Packs/FetchIndicatorsFromFile/Scripts/FetchIndicatorsFromFile/FetchIndicatorsFromFile.py

```diff
@@ -0,0 +1,221 @@
+
+
+def xls_file_to_parsed_indicators(file_path, sheet_name, indicator_col, type_col, starting_row):
+    indicator_list = []
+    hr_indicators_list = []
+
+    xl_woorkbook = xlrd.open_workbook(file_path)
+    if sheet_name and sheet_name != 'None':
+        xl_sheet = xl_woorkbook.sheet_by_name(sheet_name)
+
+    else:
+        xl_sheet = xl_woorkbook.sheet_by_index(0)
+
+    for row_index in range(0, xl_sheet.nrows):
+        if row_index >= starting_row:
+            indicator = xl_sheet.cell(row_index, indicator_col).value
+            indicator_type = xl_sheet.cell(row_index, type_col).value
+
+            hr_indicators_list.append({
+                "Type": indicator_type,
+                "Value": indicator
+            })
+
+            indicator_list.append({
+                'type': indicator_type,
+                'value': indicator
+            })
+
+    return indicator_list, hr_indicators_list
+
+
+def txt_file_to_indicator_list(file_path, auto_detect, default_type):
+    with open(file_path, "r") as fp:
+        file_data = fp.read()
+
+    indicator_list = []
+    hr_indicators_list = []
+
+    only_indicator_list = re.split('\n|,|, ', file_data)
+
+    for indicator in only_indicator_list:
+        if auto_detect:
+            indicator_type = detect_type(indicator)
+
+        else:
+            indicator_type = default_type
+
+        hr_indicators_list.append({
+            "Type": indicator_type,
+            "Value": indicator
+        })
+
+        indicator_list.append({
+            'type': indicator_type,
+            'value': indicator
+        })
+
+    return indicator_list, hr_indicators_list
+
+
+def detect_type(indicator):
+    """Infer the type of the indicator.
+    Args:
+        indicator(str): The indicator whose type we want to check.
+    Returns:
+        str. The type of the indicator.
+    """
+    if re.match(ipv4cidrRegex, indicator):
+        return FeedIndicatorType.CIDR
+
+    if re.match(ipv6cidrRegex, indicator):
+        return FeedIndicatorType.IPv6CIDR
+
+    if re.match(ipv4Regex, indicator):
+        return FeedIndicatorType.IP
+
+    if re.match(ipv6Regex, indicator):
+        return FeedIndicatorType.IPv6
+
+    if re.match(sha256Regex, indicator):
+        return FeedIndicatorType.SHA256
+
+    if re.match(urlRegex, indicator):
+        return FeedIndicatorType.URL
+
+    if re.match(md5Regex, indicator):
+        return FeedIndicatorType.MD5
+
+    if re.match(sha1Regex, indicator):
+        return FeedIndicatorType.SHA1
+
+    if re.match(emailRegex, indicator):
+        return FeedIndicatorType.Account
+
+    else:
+        return FeedIndicatorType.Domain
+
+
+def fetch_indicators_from_file(args):
+    file = demisto.getFilePath(args.get('entry_id'))
+    file_path = file['path']
+    file_name = file['name']
+    auto_detect = True if args.get('auto_detect') == 'True' else False
+    default_type = args.get('default_type')
+    limit = args.get("limit", 100)
```

Comment:
Not sure that we need the limit

Reply:
I'll keep the limit just in case if a user wants to use it for some reason.
But I'll lose the 100 so it can be unlimited.